Resources/views/Form/form_admin_fields.html.twig

```diff
@@ -1,18 +1,61 @@
 {% block sonata_security_roles_widget %}
 {% spaceless %}
-    <div class="editable">
-        <h4>{{ 'field.label_roles_editable'|trans({}, "SonataUserBundle") }}</h4>
-        {{ block('choice_widget') }}
+    {% set l = labelPermission | length %}
+    {% set i = 0 %}
+    <div class="role_table">
+        <table class="table">
+            <tr>
+                <th></th>
+                {% for j, label in labelPermission %}
+                    <th {{ 'data-x=' ~ i ~ ' data-y=' ~((j % l) + 1) }}> {{ label }} </th>
+                {% endfor %}
+            </tr>
+            {% set arr = {} %}
+            <tr>
+                {% set other = {} %}
+                {% if attribute(choices, 'other') is defined %}
+                    {% set other = attribute(choices, 'other') %}
+                {% endif %}
+                {% for j, child in form %}
+                {% if (child.vars.name % l == 0) %}
+                {% if child.vars.name != 0 %}
+            </tr><tr>
```

Comment:
If you can (not sure you can, I don't really understand your code, but the `</tr><tr>` part rings a very familiar bell), use [batch](http://twig.sensiolabs.org/doc/filters/batch.html), I think it might be very helpful here